Userland/Libraries/LibSemVer/SemVer.cpp

```diff
@@ -0,0 +1,305 @@
+        if (!this_numerical_identifier.has_value() && !other_numerical_identifier.has_value()) {
+            if (m_prerelease_identifiers[i] == other.m_prerelease_identifiers[i]) {
+                i++;
+                continue;
+            }
+            return m_prerelease_identifiers[i] > other.m_prerelease_identifiers[i];
+        }
+
+        // 3. Numeric identifiers always have lower precedence than non-numeric identifiers.
+        if (this_numerical_identifier.has_value() && !other_numerical_identifier.has_value())
+            return false;
+        if (!this_numerical_identifier.has_value() && other_numerical_identifier.has_value())
+            return true;
+    }
+
+    // 4. If all of the preceding identifiers are equal, larger set of pre-release fields has a higher precedence than a smaller set.
+    return m_prerelease_identifiers.size() > other.m_prerelease_identifiers.size();
+}
+
+bool SemVer::satisfies(StringView const& spec) const
+{
+    GenericLexer lexer(spec.trim_whitespace());
+    if (lexer.tell_remaining() == 0)
+        return false;
+
+    auto compare_op = lexer.consume_until<Function<bool(char)>>([](auto const& ch) {
+        return ch >= '0' && ch <= '9';
+    });
+
+    auto spec_version = MUST(from_string_view(lexer.consume_all()));
+    if (compare_op.length() == 0)
+        return is_same(spec_version, CompareType::Minor);
+    if (compare_op == "!=")
+        return !is_same(spec_version);
+
+    // Just match the version string
+    // Excluding prerelease and buildmetadata
+    if (compare_op == "="sv)
+        return is_same(spec_version, CompareType::Patch);
+
+    if (compare_op == "=="sv)
+        return is_same(spec_version);
+
+    // Current version is greater than spec
+    if (compare_op == ">"sv)
+        return is_greater_than(spec_version);
+    if (compare_op == "<"sv)
+        return is_lesser_than(spec_version);
+    if (compare_op == ">="sv)
+        return is_same(spec_version) || is_greater_than(spec_version);
+    if (compare_op == "<="sv)
+        return is_same(spec_version) || is_lesser_than(spec_version);
+
+    return false;
+}
+
+ErrorOr<SemVer> from_string_view(StringView const& version, char m_number_separator)
+{
+    using namespace regex;
+    if (version.count(m_number_separator) < 2)
+        return AK::Error::from_string_view("Insufficient version separator"sv);
+
+    if (version.count('+') > 1)
+        return AK::Error::from_string_view("Build metadata must be defined once"sv);
+
+    GenericLexer lexer(version.trim_whitespace());
+    if (lexer.tell_remaining() == 0)
+        return AK::Error::from_string_view("Version string is empty"sv);
+
+    // Parse the normal version parts
+    // https://semver.org/#spec-item-2
+    auto version_part = lexer.consume_until(m_number_separator).to_uint<u64>();
+    if (!version_part.has_value())
+        return AK::Error::from_string_view("Major version is not numeric"sv);
+    auto v_major = version_part.value();
```

Comment:
We can afford those 6 additional bytes if it gives us less overly abbreviated variable names.

```suggestion
    auto version_major = version_part.value();
```